File: apps/dotnet/Functions/TestErrorFunction.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;

namespace AzureFunctionsGuide.Functions;

/// <summary>
/// Controlled error test — catches and returns error details.
/// Route: GET /api/testerror
/// </summary>
public class TestErrorFunction
{
    private readonly ILogger<TestErrorFunction> _logger;

    public TestErrorFunction(ILogger<TestErrorFunction> logger)
    {
        _logger = logger;
    }

    [Function("testError")]
    public IActionResult Run(
        [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "testerror")] HttpRequest req)
    {
        _logger.LogInformation("Test error endpoint requested");

        try
        {
            throw new InvalidOperationException("Simulated error for testing");
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "Caught simulated error: {Message}", ex.Message);

            return new OkObjectResult(new
            {
                error = "Handled exception",
                type = ex.GetType().Name,
                message = ex.Message
            });
        }
    }
}

File: apps/java/src/main/java/com/functions/TestErrorFunction.java
package com.functions;

import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.*;
import java.util.Optional;

/**
 * Test error endpoint — throws intentional exception for error handling validation.
 */
public class TestErrorFunction {

    @FunctionName("testError")
    public HttpResponseMessage run(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "testerror")
            HttpRequestMessage<Optional<String>> request,
            final ExecutionContext context) {

        context.getLogger().severe("Intentional error for testing");
        throw new RuntimeException("Intentional test error from Java function");
    }
}
